Kernel/Syscalls/unveil.cpp

```diff
@@ -90,6 +83,28 @@ int Process::sys$unveil(Userspace<const Syscall::SC_unveil_params*> user_params)
         }
     }
 
+    // Now, let's try and resolve the path and obtain custody of the inode on the disk, and if not, bail out with
+    // the error from resolve_path_without_veil()
+    // However, if the user specified unveil() with "c" permissions, we don't set errno if ENOENT is encountered,
+    // because they most likely intend the program to create the file for them later on.
+    // If this case is encountered, the parent node of the path is returned and the custody of that inode is used instead.
+    RefPtr<Custody> custody_on_error; // Parent inode in case of ENOENT
+    auto custody_or_error = VFS::the().resolve_path_without_veil(path.value(), root_directory(), &custody_on_error);
+    if (custody_or_error.is_error())
+        if (!(new_permissions & UnveiledPath::Access::CreateOrRemove) && custody_or_error.error() == -ENOENT)
+            // FIXME Should this be EINVAL?
+            return custody_or_error.error();
+
+    RefPtr<Custody> custody;
+    if (custody_or_error.is_error())
```

Comment:
@awesomekling I feel like this could probably be a bit neater, but I can't use a ternary due to one being a `RefPtr` and the other being a `NonNullRefPtr`